### ISSUE TITLE
Multi port support

### ISSUE THREAD

**Issue by gbiagomba** [drwetter/testssl.sh#1388]
Please make sure that you provide enough information so that we understand what your issue is about.

1. uname -a
   Linux kali 5.2.0-kali3-amd64 #1 SMP Debian 5.2.17-1kali2 (2019-10-17) x86_64 GNU/Linux
2. testssl version from the banner: testssl.sh -b 2>/dev/null | head -4 | tail -2
   testssl       3.0rc5 from https://testssl.sh/dev/
3. git log | head -1 (if running from git repo)
   N/A
4. openssl version used by testssl.sh: testssl.sh -b 2>/dev/null | awk -F':' '/openssl/ { print $2}'
   /root/Downloads/testssl-3.0rc5/testssl.sh-3.0rc5/bin/openssl.Linux.x86_64
5. steps to reproduce: testssl.sh or docker command line, if possible incl. host
   N/A
6. what exactly was happening, output is needed
   This is a request to include a port flag that support multiple ports
7. what did you expect instead?
   I know the default work flow is hostname:443 and if you dont specify a port it will default to 443. Furthermore, I know I can point the file a gnmap file to get the request above. But I am asking for if/when I want to test SSL (no pun intended lol) across multiple hosts and multiple ports. It would be great if a port flag was added where I can specify the ports I want to check

**Comment by drwetter** [comment]
From the design perspective testssl.sh does one thing good: it's a scanner for TLS/SSL related things. It doesn't do port scanning. It would need to though when what you suggested would be implemented. Also then it would need to distinguish not only between an open and closed port. It would also need to distinguish between a just open port, an open port with pure TLS/SSL and an open/port which needs a defined STARTTLS handshake upfront. To do that properly it requires some work which I'd like rather to spend on different things.

As you pointed out you can use nmap to get an gnmap file. This should be the workflow number 1.

**Comment by gbiagomba** [comment]
But couldn't one place a net-cat command or a quick socket check (if you dont want to use nc as a dependency) to see if the port is live? then if it is try to run testssl against it? It seems pretty good at determining that a host isnt running https sp it prompts me to stop scanning. Furthermore the gnmap approach isnt perfect, I have had a few instances were I had to rerun it against a target for a port that was missed.

Anywho, I figured it would not hurt if I asked -- Wishful thinking I suppose
